File: implicit-and-streams(lab4)/src/main/java/com/epam/employee_salary/scala/Seniority.scala
package com.epam.employee_salary.scala

object SENIORITY extends Enumeration {
  val JUNIOR, MIDDLE, SENIOR = Value

  def getSeniority(employee: Employee):Value = employee match {
    case e: Employee if e.salary > 21000 => SENIORITY.SENIOR
    case e: Employee if e.salary > 14000 => SENIORITY.MIDDLE
    case e: Employee => SENIORITY.JUNIOR
  }
}

File: implicit-and-streams(lab4)/src/main/java/com/epam/employee_salary/java/Seniority.java
package com.epam.employee_salary.java;

import lombok.RequiredArgsConstructor;

import java.util.Arrays;

public enum Seniority {
    JUNIOR(0, 14000),
    MIDDLE(14000,21000),
    SENIOR(21000,Integer.MAX_VALUE);


    private final int min;
    private final int max;

    Seniority(int min, int max) {
        this.min = min;
        this.max = max;
    }


    public static Seniority getBySalary(double salary) {
        return Arrays.stream(values()).filter(seniority -> seniority.min <= salary && seniority.max > salary).findAny().get();

    }


}
